src/typescript/frontend/src/hooks/use-is-user-geoblocked.ts

```diff
@@ -0,0 +1,15 @@
+import { useQuery } from "@tanstack/react-query";
+import { isUserGeoblockedServerAction } from "utils/server/geoblocked";
+
+const useIsUserGeoblocked = () => {
+  const { data } = useQuery({
+    queryKey: ["geoblocked"],
+    queryFn: () => isUserGeoblockedServerAction(),
+    staleTime: Infinity,
```

Comment:
I think this should be something between 2 and 7 days. Ideally, when data is stale, it would return the stale data while new data is getting loaded, to avoid flashing the geo-blocked banner.

Reply:
Addressed in 20575d7e

Reply:
Updated this to not show the banner unless they're explicitly geoblocked by adding an optional param to the hook that's false by default. That is, without specifying that you want to know if they're explicitly geoblocked (no nullish coalescing), we pass `{ explicitlyGeoblocked = true }` to the hook.

Otherwise, the data returned is `data ?? true` (assume they're geoblocked by default).

This lets us configure the banner to only show up if the user is explicitly geoblocked, but still block the user by default in other components without having to refactor those hook invocations.

cd7aab12cb0be9e1898ebae42b701e9e4fb8b2b4